File: gym/1400/C1854.py
import sys 
input = sys.stdin.readline 

def solve():
    s = input().strip()
    m = int(input())
    l = input().strip()
    r = input().strip()
    p = 0
    for i in range(m):
        q = p
        for d in range(int(l[i]), int(r[i]) + 1):
            pos = s.find(str(d), p)
            if pos == -1: 
                print("YES") 
                return 
            q = max(q, pos+1)
    print("NO")

t = int(input())
for _ in range(t):
    solve()

File: gym/1400/C1854.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
void solve() {
    string s; cin >> s;
    int m; cin >> m;
    string l, r; cin >> l >> r;
    int p = 0;
    for (int i = 0; i < m; ++i) {
        int q = p;
        for (char x = l[i]; x <= r[i]; x++) {
            if (s.find(x, p) == -1) {
                cout << "YES\n";
                return;
            }
            q = max(q, int(s.find(x, p) ) + 1);
        }
        p = q;
    }
    cout << "NO\n";
}
int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr);
    int t; cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}